src/components/InputCheckBox.tsx

```diff
@@ -0,0 +1,28 @@
+import './InputCheckBox.css';
+
+type InputCheckBoxProps = {
+  id: string;
+  checked: boolean;
+  onChange: (e : React.FormEvent<HTMLInputElement>) => void;
+  label?: string;
+  required?: boolean;
+}
+
+const InputCheckBox = ({id, checked, onChange, label, required}: InputCheckBoxProps) => {
+  return (
+    <div className='inputCheckBox'>
+      <input
+        className='inputCheckBox__input'
+        type="checkbox"
+        id={id}
+        checked={checked}
+        onChange={onChange}
+        />
+      <label htmlFor={id} className={required ? 'inputCheckBox__title inputCheckBox__title__required' : 'inputCheckBox__title'}>
+        {label}{required && '*'}
+      </label>
+    </div>
+  )
+}
```

Comment:
label의 className을 결정하는 부분이 inputBox와 inputCheckBox에서 반복됩니다. 이 부분을 필수 className과 플래그에 따라 추가여부가 정해지는 className, 그리고 플래그를 인자로 받아서 클래스 네임을 리턴해주는 함수로 바꿔보면 어떨까요?

```suggestion
const InputCheckBox = ({id, checked, onChange, label, required}: InputCheckBoxProps) => {
  const labelClassName = getClassNameByRequired('inputCheckBox__title', 'inputCheckBox__title__required', required);
  return (
    <div className='inputCheckBox'>
      <input
        className='inputCheckBox__input'
        type="checkbox"
        id={id}
        checked={checked}
        onChange={onChange}
        />
      <label htmlFor={id} className={labelClassName}>
        {label}{required && '*'}
      </label>
    </div>
  )
}
```
```js
const getClassNameByRequired = (core:string, suffix:string, required:boolean) => {
  return (required) ?  `${core} ${suffix}` : `${core}`
}
```